DESIGN.md

```diff
@@ -0,0 +1,412 @@
+# COSE Golang Library
+
+## Objectives
+
+The objectives of the `go-cose` library of this doc are
+
+- Implement `COSE_Sign` and `COSE_Sign1` as specified in [RFC8152](https://datatracker.ietf.org/doc/html/rfc8152).
+- Make the implementation secure and efficient.
+- Provide signing and verification based on golang built-in crypto libraries.
+- Provide extensibility so that other implementation can be used for signing and verification.
+  - This objective implies remote signing.
+- Be golang native.
+  - Write documentation and examples for `godoc`.
+  - Test using `go test` with both positive and negative test cases.
+
+## Implementation Gaps
+
+Here are the gaps between the current implementation and the objectives:
+
+- The current implementation implements `COSE_Sign` and `COSE_Sign1`. However, the verification process does not follow [RFC8152](https://datatracker.ietf.org/doc/html/rfc8152) and causes [issue #7](https://github.com/veraison/go-cose/issues/7). There are also implementation errors in signing and verification (see [issue #8](https://github.com/veraison/go-cose/issues/8)).

```

Comment:
I think the tense of the sentence should now reflect past tense with the implementation, example like earlier implementation. It should also say, had caused issues like #7 and #8 which is been fixed in the current implementation.

Reply:
@shizhMSFT, can you make this minor wording change? 
Suggested change:

Prior versions of go-cose implemented `COSE_Sign` and `COSE_Sign1`. However, the verification process did not follow [RFC8152](https://datatracker.ietf.org/doc/html/rfc8152) which had caused issues like [#7](https://github.com/veraison/go-cose/issues/7) and [#8](https://github.com/veraison/go-cose/issues/8) which have been fixed in the current implementation.

Reply:
updated in 6ef3dd4